vterm/stdout.go

```diff
@@ -99,9 +99,25 @@ func (v *VTerm) ProcessStdout(input *bufio.Reader) {
 				}
 
 				v.RedrawWindow() // FIXME inefficient
-			case ecma48.DCH: // delete characters
+			case ecma48.ECH: // erase characters - makes characters to the right blank
 				if x.N > v.w-v.Cursor.X {
-					x.N = v.w - v.Cursor.X // FIXME: verify that we don't need +/- 1
```

Comment:
The generated diff is weird here, but we know that we need `-1` because the vterm docs specify that `v.Cursor.X < v.w`, so if `v.Cursor.X == v.w - 1` (we're at the end), then `v.w - v.Cursor.X - 1 == 0`, meaning that if we're at the end of a line, don't move forward more